pandas/_testing/asserters.py

```diff
@@ -686,7 +684,7 @@ def assert_numpy_array_equal(
     left,
     right,
     strict_nan=False,
-    check_dtype=True,
+    check_dtype: bool | Literal["equiv"] = True,
```

Comment:
If an argument has a default value but not annotation, pyright assumes that the type of the default value is the only accepted type (there are exceptions for `None`).

Reply:
Based on the docstrings, `assert_almost_equal` is the only function edited here that takes `"equiv"` while the others should only take True/False. So shouldn't these just be `bool`?

Reply:
`assert_almost_equal` calls `assert_frame_equal` and `assert_series_equal`, potentially with `"equiv"`. `assert_series_equal` calls then `assert_extension_array_equal` and `assert_numpy_array_equal`. All of them work with `"equiv"` because most of them just check for the truthiness of `check_dtype` (a non-empty string is True).

Could update the doc-string (but on the other side non of those functions raises errors on other strings), leave as is, or deprecate `equiv` for `check_dtype` (if are no future intentions to actually distinguish between True and equiv for `check_dtype`).

Reply:
I see thanks for the explanation. I think eventually it's would be better if `equiv` was documented and that these functions were made stricter to not just check truthiness of strings, but that's for another PR.